css/amp.css

```diff
@@ -153,6 +153,14 @@ html.i-amphtml-ios-embed.i-amphtml-ios-overscroll > #i-amphtml-wrapper {
   position: relative;
 }
 
+/**
+ * Children of responsive elements should not be able to influence the height
+ * which they do when they are statically positioned.
+ */
+.i-amphtml-layout-responsive > :not(i-amphtml-sizer) {
```

Comment:
ok now I understand what this is doing but now curious why the existing
```
.i-amphtml-notbuilt:not(.i-amphtml-layout-container) > * {
  display: none;
}
```
doesn't handle this already? (Looks like it is meant to handle the "stubbed-but-not-built" case)

Before carousel is built, all its children should be hidden and therefore not cause any expansion of `<amp-carousel> regardless of layout, by the time we remove `i-amphtml-notbuilt`, it should be done building and its children should already be properly contained within an absolute parent. Obviously this is not happening, so maybe the right fix lies there.